src/scripts/main.js

```diff
@@ -1,3 +1,15 @@
 'use strict';
 
 // write code here
+
+const listaUl = document.querySelector('ul');
+const listaLi = listaUl.querySelectorAll('li');
+
+const posortowanaLista = [...listaLi].sort((a, b) => {
+  const salaryA = Number(a.getAttribute('data-salary').replace(/[$,]/g, ''));
+  const salaryB = Number(b.getAttribute('data-salary').replace(/[$,]/g, ''));
+
+  return salaryB - salaryA;
+});
+
+posortowanaLista.forEach((li) => listaUl.appendChild(li));
```

Comment:
Appending the 'li' elements back to the 'ul' after sorting is a good approach, but you can optimize the DOM manipulation by using 'DocumentFragment' or performing the append operation outside of the loop to minimize reflows and repaints.